src/transformStateWithClones.js

```diff
@@ -8,6 +8,44 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const transformedClones = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const clone = { ...stateCopy };
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          clone[data] = action.extraData[data];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          for (const data in clone) {
+            if (data === key) {
+              delete clone[data];
+            }
+          }
+        }
```

Comment:
no needs to use a loop twice
```suggestion
        for (const key of action.keysToRemove) {
          delete stateCopy[key];
        }
```